common/queue/src/main/java/org/thingsboard/server/queue/kafka/TbKafkaConsumerStatsService.java

```diff
@@ -0,0 +1,145 @@
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+@ConditionalOnProperty(prefix = "queue", value = "type", havingValue = "kafka")
+public class TbKafkaConsumerStatsService {
+    private final Set<String> monitoredGroups = ConcurrentHashMap.newKeySet();
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private final TbKafkaSettings kafkaSettings;
+    private final TbKafkaConsumerStatisticConfig statsConfig;
+
+    private AdminClient adminClient;
+    private Consumer<String, byte[]> consumer;
+    private ScheduledExecutorService statsPrintScheduler;
+
+    @PostConstruct
+    public void init() {
+        if (!statsConfig.getEnabled()) {
+            return;
+        }
+        this.adminClient = AdminClient.create(kafkaSettings.toAdminProps());
+        this.statsPrintScheduler = Executors.newSingleThreadScheduledExecutor(ThingsBoardThreadFactory.forName("kafka-consumer-stats"));
+
+        Properties consumerProps = kafkaSettings.toConsumerProps();
+        consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "consumer-stats-loader-client");
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "consumer-stats-loader-client-group");
+        this.consumer = new KafkaConsumer<>(consumerProps);
+
+        startLogScheduling();
+    }
+
+    private void startLogScheduling() {
+        Duration timeoutDuration = Duration.ofMillis(statsConfig.getKafkaResponseTimeoutMs());
+        statsPrintScheduler.scheduleWithFixedDelay(() -> {
+            for (String groupId : monitoredGroups) {
+                try {
+                    Map<TopicPartition, OffsetAndMetadata> groupOffsets = adminClient.listConsumerGroupOffsets(groupId).partitionsToOffsetAndMetadata()
+                            .get(statsConfig.getKafkaResponseTimeoutMs(), TimeUnit.MILLISECONDS);
+                    Map<TopicPartition, Long> endOffsets = consumer.endOffsets(groupOffsets.keySet(), timeoutDuration);
+
+                    List<GroupTopicStats> consumerGroupStats = new ArrayList<>();
+                    for (TopicPartition topicPartition : groupOffsets.keySet()) {
+                        Long endOffset = endOffsets.get(topicPartition);
+                        long committedOffset = groupOffsets.get(topicPartition).offset();
+                        GroupTopicStats groupTopicStats = GroupTopicStats.builder()
+                                .topic(topicPartition.topic())
+                                .partition(topicPartition.partition())
+                                .committedOffset(committedOffset)
+                                .endOffset(endOffset)
+                                .lag(endOffset - committedOffset)
+                                .build();
+                        consumerGroupStats.add(groupTopicStats);
+                    }
+                    log.debug("[{}] Stats: [{}].", groupId, mapper.writeValueAsString(consumerGroupStats));
```

Comment:
This should be "info".

Reply:
Also, please log the data in a "csv" like form, not a JSON.

Reply:
Pls check if that's the correct format for the log